docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,497 @@
+
+> [!IMPORTANT]
+> This shall not be a drop-in replacement for [`k3s`](https://k3s.io) or [`kubeadm`](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm) (that would be too far away from Gardener's mission statement, TCO goals, and current implementation), but we want to offer this new type of shoot clusters as a separate flavor to enable on-prem use-cases that cannot be served with Gardener today.
+> By allowing the remote management through Gardener, "autonomous" clusters will meet our mission statement (e.g., in regards to low TCO) while operating without the need of a seed cluster.
+
+Such autonomous shoot clusters are meant to operate autonomously, but not to exist completely independently of Gardener.
+Hence, after initial creation, they shall be connected to an existing Gardener system such that the established cluster management functionality via the `Shoot` API can be applied.
+
+Furthermore, autonomous shoot clusters are not considered a replacement or alternative for regular shoot clusters.
+They should be only used for special use-cases or requirements as their creation time will be longer and their costs will be higher.
+In this light, a high cluster creation/deletion churn rate is neither expected nor in scope.
+
+Gardener's APIs and cluster topology are defined in a way such that seed clusters are a separate "security domain" and not accessible to end-users.
+With autonomous shoot clusters, seed clusters do not exist anymore, hence, running a `gardenlet` and certain seed system components (like extensions) within the end-user's infrastructure requires rework and adaptation of some of today's concepts.
+As operating autonomous shoot clusters is a challenge big enough on its own, we'll start with focusing only on how Gardener operators can create and connect autonomous shoot clusters.
+Allowing "untrusted" end-users to create such clusters is out of scope of this GEP and considered to be future work ([see below](#autonomous-shoot-clusters-for-end-users)).
+
+## Motivation
+
+Gardener is a great tool for creating and managing clusters with very low TCO.
+Part of this success is its architecture, i.e., running control planes in seed clusters under joint supervision by Gardener and Kubernetes (a.k.a. Kubeception deployment model).
+
+However, there is sufficient pull to also find a way to create autonomous shoot clusters using Gardener (e.g., for the edge or for temporarily air-gapped scenarios) where the control plane must run side-by-side and cannot run in seed clusters.
+We want to establish Gardener in (new) environments where clusters cannot run their control plane "somewhere else", but where it needs to be co-located with the worker nodes, e.g., to avoid costly or exposed network traffic, firewall the entire cluster off, rely on the control plane while the cluster is air-gapped/has no internet access, generally avoid runtime dependencies, fulfill compliance obligations, etc.
```

Comment:
Could you please go into more details why this is related to this GEP? My assumption is that we reuse the existing means, e.g. image vector overwrites. Hence, some of your concerns should already be automatically be addressed. Others, e.g. DNS and certificates, I do not understand, yet.

Reply:
My point is that whoever installs gardener in a fenced environment may need to replicate the entire image vector into own image registry and this registry must be accessible from within the fenced environment. Of course, this is obvious for experienced users of gardener, but for new users it is most probably not, this is why I think such prerequisites should be properly documented. 

With regards to DNS and SSL, Gardener is heavily using DNS hostnames to advertise certain endpoints like the k8s API server and the observability stack (plutono, prometheus, vali, etc.). In the past there was option plain IP addresses to be used, but as far as I know this option was removed. For some of the DNS names SSL/TLS certificates are requested by external certificate provider instead of using self generated/signed certificates.

In the fenced environment I doubt services like AWS Route 53 or Let's Encrypt to be available or allowed, hence respective alternatives will be required. 

I am asking such requirements to be properly documented so that the environment to be correctly prepared before somebody to try to install gardener.

Reply:
I absolutely agree that we need documentation how to setup a Gardener landscape in a fenced environment. However, I do not see this documentation as part of the GEP for autonomous shoot clusters.

You can already setup a Gardener landscape in a fenced environment today. It requires to have a root kubernetes cluster, but that can be done with `kubeadm`, `k3s` or some other distribution. Autonomous shoot clusters aims at simplifying this part by removing the necessity of a root kubernetes managed by some other software. It is no precondition for setting up a fenced environment, though.

If you do not mind I would propose to create a corresponding issue on gardener/documentation where such documentation would belong from my point of view.
WDYT?

Reply:
I agree to some extend with you that even today Gardener can be installed behind firewall and such difficulties have be resolved, but as autonomous clusters provisioned by Gardener targets fenced environments as problem area that they can solve, from this point of view such a small task to properly document the requirements to run Gardener or just autonomous cluster behind a firewall could be implemented as part of the GEP.

Reply:
Just to be on the same page: Would it be fine for you if we add a section describing the problem statement in the future work chapter? The documentation itself should surely not be included in the GEP, but it can be created as part of the GEP implementation. I understand your comment that you want the task to be included in the GEP. My preference would be to include it in the future work.
WDYT? Feel free to propose other places where you want to have it added instead if the proposal does not work for you.

Reply:
Sounds good to me.

Reply:
Added section in https://github.com/gardener/gardener/pull/10536/commits/99eefb454c4ebd6701eab79312bee3ddd8ae1f56.
PTAL.